Description: Fix a typo in filetags.
Forwarded: https://gitorious.org/bangarang/bangarang/merge_requests/2258
Author: Ryan Kavanagh <ryanakca@kubuntu.org>
Origin: vendor
Reviewed-By: Ryan Kavanagh <ryanakca@kubuntu.org>
Last-Update: 2011-05-27
Index: bangarang/src/platform/utilities/filetags.cpp
===================================================================
--- bangarang.orig/src/platform/utilities/filetags.cpp	2011-05-27 08:02:28.602029205 -0400
+++ bangarang/src/platform/utilities/filetags.cpp	2011-05-27 08:02:45.802114494 -0400
@@ -717,7 +717,7 @@
     genreDictionary[64] = "Native American";
     genreDictionary[65] = "Cabaret";
     genreDictionary[66] = "New Wave";
-    genreDictionary[67] = "Psychadelic";
+    genreDictionary[67] = "Psychedelic";
     genreDictionary[68] = "Rave";
     genreDictionary[69] = "Showtunes";
     genreDictionary[70] = "Trailer";
